From a28958630e4c4d6cd3c19dbdbe6ad54b0d1b9ca5 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Sun, 18 Oct 2026 01:33:30 -0700
Subject: [PATCH] chain: brand daemon and RPC client targets as Lotus

Rename daemon and CLI targets from bitcoind/bitcoin-cli to lotusd/lotus-cli
for the Lotus V2 network.

Original-Commit: 661cb3bed1e31b9f1f241181d1a94ad250d75549
Original-Patch: v2/003
Split-Concern: whole

Historical compilation prerequisites are tracked separately in daemon/patch-audit;
this provenance-preserving split does not imply standalone build validation.

diff --git a/doc/man/CMakeLists.txt b/doc/man/CMakeLists.txt
index 4a7c0341df07cc5bb3f20ef49c88c340e1bc2dbd..8cc552c9b069733c5fe536508f82fc78973aecae 100644
--- a/doc/man/CMakeLists.txt
+++ b/doc/man/CMakeLists.txt
@@ -18,11 +18,11 @@ function(generate_manpages)
 				"${CMAKE_SOURCE_DIR}/cmake/utils/log-and-print-on-failure.sh"
 				"${CMAKE_BINARY_DIR}/doc/man/gen-manpage-${TARGET}.log"
 				"${CMAKE_SOURCE_DIR}/doc/man/gen-manpages.sh"
-				"$<TARGET_FILE:bitcoind>"
+				"$<TARGET_FILE:lotusd>"
 				"$<TARGET_FILE:${TARGET}>"
 				"${MAN_PAGE}"
 			DEPENDS
-				bitcoind
+				lotusd
 				"${TARGET}"
 		)
 		add_custom_target(doc-manpage-${TARGET}
diff --git a/doc/rpc/CMakeLists.txt b/doc/rpc/CMakeLists.txt
index d2c26d3cadac6a31e7bec0b16052d22c469f8301..4a4bcba41db57a84b51a3a54b02a53a78b9f54f9 100644
--- a/doc/rpc/CMakeLists.txt
+++ b/doc/rpc/CMakeLists.txt
@@ -6,12 +6,12 @@ if(BUILD_CLI)
 		COMMAND
 			"${CMAKE_CURRENT_SOURCE_DIR}/gen-rpc-docs.sh"
 			"${CMAKE_CURRENT_SOURCE_DIR}/generate.go"
-			"$<TARGET_FILE:bitcoind>"
-			"$<TARGET_FILE:bitcoin-cli>"
+			"$<TARGET_FILE:lotusd>"
+			"$<TARGET_FILE:lotus-cli>"
 		WORKING_DIRECTORY "${CMAKE_BINARY_DIR}"
 		DEPENDS
-			bitcoind
-			bitcoin-cli
+			lotusd
+			lotus-cli
 	)
 endif()
 
diff --git a/electrum/CMakeLists.txt b/electrum/CMakeLists.txt
index 7ae6dbcd7bd8a7ac747c0baeac7fd4921984281f..6985f3fd6ba1709514893e18f064683d52c92e69 100644
--- a/electrum/CMakeLists.txt
+++ b/electrum/CMakeLists.txt
@@ -18,10 +18,10 @@ if(BUILD_CHRONIK)
 	add_custom_target(check-electrum-functional
 		COMMENT "Run Electrum ABC functional tests..."
 		COMMAND
-			"BITCOIND=${CMAKE_BINARY_DIR}/src/bitcoind"
+			"BITCOIND=${CMAKE_BINARY_DIR}/src/lotusd"
 			"${Python_EXECUTABLE}" ./test_runner.py functional
 		DEPENDS
-			bitcoind
+			lotusd
 			${CMAKE_CURRENT_BINARY_DIR}/test_runner.py
 	)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 72274d434ec1ca3f6899a8886ca6ec9f6faf0c7e..cf365406c61c99655a5d92267016e0ca678dce6e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Copyright (c) 2017 The Bitcoin developers
 
-project(bitcoind)
+project(lotusd)
 
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
@@ -16,7 +16,7 @@ endforeach()
 
 option(BUILD_WALLET "Activate the wallet functionality" ON)
 option(BUILD_ZMQ "Activate the ZeroMQ functionalities" ON)
-option(BUILD_CLI "Build bitcoin-cli" ON)
+option(BUILD_CLI "Build lotus-cli" ON)
 option(BUILD_TX "Build bitcoin-tx" ON)
 option(BUILD_QT "Build bitcoin-qt" ON)
 option(BUILD_SEEDER "Build bitcoin-seeder" ON)
@@ -803,21 +803,21 @@ if(BUILD_SEEDER)
 	add_subdirectory(seeder)
 endif()
 
-# bitcoin-cli
+# lotus-cli
 if(BUILD_CLI)
-	add_executable(bitcoin-cli bitcoin-cli.cpp)
-	generate_windows_version_info(bitcoin-cli
+	add_executable(lotus-cli bitcoin-cli.cpp)
+	generate_windows_version_info(lotus-cli
 		DESCRIPTION "JSON-RPC client for ${PACKAGE_NAME}"
 	)
 
-	target_link_libraries(bitcoin-cli common rpcclient)
-	link_event(bitcoin-cli event)
+	target_link_libraries(lotus-cli common rpcclient)
+	link_event(lotus-cli event)
 
-	add_to_symbols_check(bitcoin-cli)
-	add_to_security_check(bitcoin-cli)
+	add_to_symbols_check(lotus-cli)
+	add_to_security_check(lotus-cli)
 
-	install_target(bitcoin-cli)
-	install_manpages(bitcoin-cli)
+	install_target(lotus-cli)
+	install_manpages(lotus-cli)
 endif()
 
 # bitcoin-tx
@@ -1009,17 +1009,17 @@ if(BUILD_CHAINSTATE)
 	add_to_security_check(bitcoin-chainstate)
 endif()
 
-# bitcoind
-add_executable(bitcoind bitcoind.cpp)
-target_link_libraries(bitcoind server)
-generate_windows_version_info(bitcoind
-	DESCRIPTION "Bitcoin node with a JSON-RPC server"
+# lotusd
+add_executable(lotusd bitcoind.cpp)
+target_link_libraries(lotusd server)
+generate_windows_version_info(lotusd
+	DESCRIPTION "Lotus node with a JSON-RPC server"
 )
-add_to_symbols_check(bitcoind)
-add_to_security_check(bitcoind)
+add_to_symbols_check(lotusd)
+add_to_security_check(lotusd)
 
-install_target(bitcoind)
-install_manpages(bitcoind)
+install_target(lotusd)
+install_manpages(lotusd)
 
 # Bitcoin-qt
 if(BUILD_QT)
diff --git a/src/bitcoin-cli.cpp b/src/bitcoin-cli.cpp
index 91cd1918995959e33c4bdf43e187a93d1e5f0660..53717d77087707a4be3554aba3d775fb7d932747 100644
--- a/src/bitcoin-cli.cpp
+++ b/src/bitcoin-cli.cpp
@@ -82,7 +82,7 @@ static void SetupCliArgs(ArgsManager &argsman) {
             "arguments are number of blocks to generate (default: %s) and "
             "maximum iterations to try (default: %s), equivalent to RPC "
             "generatetoaddress nblocks and maxtries arguments. Example: "
-            "bitcoin-cli -generate 4 1000",
+            "lotus-cli -generate 4 1000",
             DEFAULT_NBLOCKS, DEFAULT_MAX_TRIES),
         ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
     argsman.AddArg(
@@ -160,7 +160,7 @@ static void SetupCliArgs(ArgsManager &argsman) {
     argsman.AddArg(
         "-rpcwallet=<walletname>",
         "Send RPC for non-default wallet on RPC server (needs to exactly match "
-        "corresponding -wallet option passed to bitcoind). This changes the "
+        "corresponding -wallet option passed to lotusd). This changes the "
         "RPC endpoint used, e.g. http://127.0.0.1:8332/wallet/<walletname>",
         ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
 }
@@ -211,14 +211,14 @@ static int AppInitRPC(int argc, char *argv[]) {
             strUsage += FormatParagraph(LicenseInfo());
         } else {
             strUsage += "\n"
-                        "Usage:  bitcoin-cli [options] <command> [params]  "
+                        "Usage:  lotus-cli [options] <command> [params]  "
                         "Send command to " PACKAGE_NAME "\n"
-                        "or:     bitcoin-cli [options] -named <command> "
+                        "or:     lotus-cli [options] -named <command> "
                         "[name=value]...  Send command to " PACKAGE_NAME
                         " (with named arguments)\n"
-                        "or:     bitcoin-cli [options] help                "
+                        "or:     lotus-cli [options] help                "
                         "List commands\n"
-                        "or:     bitcoin-cli [options] help <command>      Get "
+                        "or:     lotus-cli [options] help <command>      Get "
                         "help for a command\n";
 
             strUsage += "\n" + gArgs.GetHelpMessage();
@@ -510,7 +510,7 @@ public:
 
         const UniValue &networkinfo{batch[ID_NETWORKINFO]["result"]};
         if (networkinfo["version"].getInt<int>() < 230000) {
-            throw std::runtime_error("-netinfo requires bitcoind server to be "
+            throw std::runtime_error("-netinfo requires lotusd server to be "
                                      "running v0.23.0 and up");
         }
 
@@ -813,7 +813,7 @@ static UniValue CallRPC(BaseRequestHandler *rh, const std::string &strMethod,
         }
         throw CConnectionFailed(
             strprintf("Could not connect to the server %s:%d%s\n\nMake sure "
-                      "the bitcoind server is running and that you are "
+                      "the lotusd server is running and that you are "
                       "connecting to the correct RPC port.",
                       host, port, responseErrorMessage));
     } else if (response.status == HTTP_UNAUTHORIZED) {
@@ -919,7 +919,7 @@ static void ParseError(const UniValue &error, std::string &strPrint,
         if (err_code.isNum() &&
             err_code.getInt<int>() == RPC_WALLET_NOT_SPECIFIED) {
             strPrint += "\nTry adding \"-rpcwallet=<filename>\" option to "
-                        "bitcoin-cli command line.";
+                        "lotus-cli command line.";
         }
     } else {
         strPrint = "error: " + error.write();
diff --git a/src/bitcoind.cpp b/src/bitcoind.cpp
index 16d1201338e189f0927486a505e5669876663aff..8efba0cb73a164655919e519e8b8b380d0d16d35 100644
--- a/src/bitcoind.cpp
+++ b/src/bitcoind.cpp
@@ -162,7 +162,7 @@ static bool AppInit(NodeContext &node, int argc, char *argv[]) {
         if (args.IsArgSet("-version")) {
             strUsage += FormatParagraph(LicenseInfo()) + "\n";
         } else {
-            strUsage += "\nUsage:  bitcoind [options]                     "
+            strUsage += "\nUsage:  lotusd [options]                     "
                         "Start " PACKAGE_NAME "\n";
             strUsage += "\n" + args.GetHelpMessage();
         }
@@ -201,12 +201,12 @@ static bool AppInit(NodeContext &node, int argc, char *argv[]) {
             if (!IsSwitchChar(argv[i][0])) {
                 return InitError(Untranslated(
                     strprintf("Command line contains unexpected token '%s', "
-                              "see bitcoind -h for a list of options.",
+                              "see lotusd -h for a list of options.",
                               argv[i])));
             }
         }
 
-        // -server defaults to true for bitcoind but not for the GUI so do this
+        // -server defaults to true for lotusd but not for the GUI so do this
         // here
         args.SoftSetBoolArg("-server", true);
         // Set this early so that parameter interactions go to console
@@ -311,7 +311,7 @@ int main(int argc, char *argv[]) {
 
     SetupEnvironment();
 
-    // Connect bitcoind signal handlers
+    // Connect lotusd signal handlers
     noui_connect();
 
     return (AppInit(node, argc, argv) ? node.exit_status.load() : EXIT_FAILURE);
diff --git a/src/clientversion.cpp b/src/clientversion.cpp
index 8df98f99d2ffcdd91a59239374846efcd8b10853..b5a7f7159cb96756edc9f72d3436364250a0e5e2 100644
--- a/src/clientversion.cpp
+++ b/src/clientversion.cpp
@@ -16,7 +16,7 @@
  * for both bitcoind and bitcoin-qt, to make it harder for attackers to
  * target servers or GUI users specifically.
  */
-const std::string CLIENT_NAME("Bitcoin ABC");
+const std::string CLIENT_NAME("Lotus");
 
 #ifdef HAVE_BUILD_INFO
 #include <obj/build.h>
@@ -92,8 +92,8 @@ std::string CopyrightHolders(const std::string &strPrefix) {
 
 std::string LicenseInfo() {
     const std::string URL_SOURCE_CODE =
-        "<https://github.com/Bitcoin-ABC/bitcoin-abc>";
-    const std::string URL_WEBSITE = "<https://www.bitcoinabc.org>";
+        "<https://github.com/givelotus/lotus>";
+    const std::string URL_WEBSITE = "<https://github.com/givelotus/lotus>";
 
     return CopyrightHolders(strprintf(_("Copyright (C) %i-%i").translated, 2009,
                                       COPYRIGHT_YEAR) +
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index c11a6959c9673009ca266507bbf5ad7c3913bf69..57e437863fcab369427549eb79b9dcee021bad71 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -26,7 +26,7 @@ include(TestSuite)
 
 set(_TEST_TARGET_DEPENDS "")
 if(BUILD_CLI)
-	list(APPEND _TEST_TARGET_DEPENDS bitcoin-cli)
+	list(APPEND _TEST_TARGET_DEPENDS lotus-cli)
 endif()
 if(BUILD_WALLET)
 	list(APPEND _TEST_TARGET_DEPENDS bitcoin-wallet)
@@ -73,7 +73,7 @@ function(add_functional_test_check TARGET DESCRIPTION)
 		CUSTOM_TARGET_ARGS
 			COMMENT "Running ${DESCRIPTION}"
 			DEPENDS
-				bitcoind
+				lotusd
 				${_TEST_TARGET_DEPENDS}
 				${CMAKE_CURRENT_BINARY_DIR}/functional/test_runner.py
 			USES_TERMINAL
diff --git a/test/functional/test_framework/test_framework.py b/test/functional/test_framework/test_framework.py
index 7681219dae105f44aa08d06aab1d2b2737d6d552..427a17676191d55c26070f0b460e8bcaac2fa6e2 100644
--- a/test/functional/test_framework/test_framework.py
+++ b/test/functional/test_framework/test_framework.py
@@ -333,12 +333,12 @@ class BitcoinTestFramework(metaclass=BitcoinTestMetaClass):
         fname_bitcoind = os.path.join(
             config["environment"]["BUILDDIR"],
             "src",
-            f"bitcoind{config['environment']['EXEEXT']}",
+            f"lotusd{config['environment']['EXEEXT']}",
         )
         fname_bitcoincli = os.path.join(
             config["environment"]["BUILDDIR"],
             "src",
-            f"bitcoin-cli{config['environment']['EXEEXT']}",
+            f"lotus-cli{config['environment']['EXEEXT']}",
         )
         self.options.bitcoind = os.getenv("BITCOIND", default=fname_bitcoind)
         self.options.bitcoincli = os.getenv("BITCOINCLI", default=fname_bitcoincli)
